### PR TITLE
Multiline parentheses objectExpression in indexer expression

### PR DESCRIPTION
**Pull request by janus**
Idempotency problem when using lower MaxLineLength than default and fantomas needs to split lines

 Fixes https://github.com/fsprojects/fantomas/issues/2176

### CODE REVIEW

File: src/Fantomas.Tests/ComputationExpressionTests.fs [hunk 1]
```diff
@@ -2436,3 +2436,42 @@ async {
     return model.Prop
 }
 """
+
+[<Test>]
+let ``Idempotency problem when using lower MaxLineLength than default, 2176`` () =
```
Comment (nojaf) [inline src/Fantomas.Tests/ComputationExpressionTests.fs:2441]: This problem is not related to computation expressions, please move to `DotIndexedGetTests`.
The test name should not start with an uppercase letter. See https://github.com/fsprojects/fantomas/blob/master/CONTRIBUTING.md#unit-test

The test name itself does also not cover what the problem case is.
Rename to `multiline parentheses objectExpression in indexer expression, 2176`

File: src/Fantomas/CodePrinter.fs [hunk 2]
```diff
@@ -2453,7 +2453,12 @@ and genExpr astContext synExpr ctx =
 
             expressionFitsOnRestOfLine (short +> idx) (long +> idx)
         | DotIndexedGet (objectExpr, indexArgs) ->
-            addParenIfAutoNln objectExpr (genExpr astContext)
+            let isParen =
```
Comment (nojaf) [inline src/Fantomas/CodePrinter.fs:2456]: This is quite confusing, `isParen` is false when the expression is `SynExpr.Paren`.

**Review by nojaf — CHANGES_REQUESTED** [review]
Please add a changelog entry as well.

File: src/Fantomas.Tests/ComputationExpressionTests.fs (hunk 1 above)
Comment (janus) [reply src/Fantomas.Tests/ComputationExpressionTests.fs]: Done

File: src/Fantomas/CodePrinter.fs (hunk 2 above)
Comment (janus) [reply src/Fantomas/CodePrinter.fs:2456]: Done

File: src/Fantomas.Tests/ComputationExpressionTests.fs (hunk 1 above)
Comment (janus) [reply src/Fantomas.Tests/ComputationExpressionTests.fs]: Thanks.

**Comment by janus** [comment]
> Please add a changelog entry as well.

Thanks. Done.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,7 @@
 ### Fixed
 * SRTP or condition disappear when non-generic type is used. [#2168](https://github.com/fsprojects/fantomas/issues/2168)
 * Multiline AbstractSlot without constraints introduces newline [#2175](https://github.com/fsprojects/fantomas/issues/2175)
+* Multiline parentheses objectExpression in indexer expression, 2176 [#2176](https://github.com/fsprojects/fantomas/issues/2176)
```
Comment (nojaf) [inline CHANGELOG.md:8]: 4.7.6 was already release, you cannot add any items here.
Please use an `## [Unreleased]` section, see https://keepachangelog.com/en/1.0.0/

**Review by nojaf — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (janus) [reply CHANGELOG.md]: Done, sorry.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## [Unreleased]
+
+### Fixed
+* Multiline parentheses objectExpression in indexer expression, 2176 [#2176](https://github.com/fsprojects/fantomas/issues/2176)
```
Comment (nojaf) [inline CHANGELOG.md:6]: The issue number is being duplicated here, this is inconsistent with all the other entries.

**Review by nojaf — CHANGES_REQUESTED** [review]
Almost there

File: CHANGELOG.md (hunk 4 above)
Comment (janus) [reply CHANGELOG.md]: Done, sorry.

**Review by nojaf — APPROVED** [review]
Thank you!
